tests/core/test_core_containers.cpp

```diff
@@ -71,6 +71,48 @@ TEST_F(core_container_test, device_vector) {
     }
 }
 
+/// Test(s) for @c vecmem::device_vector
+TEST_F(core_container_test, device_nonconst_vector) {
+
+    vecmem::device_vector<int> test_vector(
+        vecmem::get_data(m_reference_vector));
+    EXPECT_EQ(test_vector.size(), m_reference_vector.size());
+    EXPECT_EQ(test_vector.empty(), m_reference_vector.empty());
+    EXPECT_TRUE(std::equal(m_reference_vector.begin(), m_reference_vector.end(),
+                           test_vector.begin()));
+    EXPECT_EQ(std::accumulate(test_vector.begin(), test_vector.end(), 0),
+              std::accumulate(test_vector.rbegin(), test_vector.rend(), 0));
```

Comment:
This is a duplication of tests we have already. I don't see the point in duplicating them.

Reply:
See comment above.